modules/candidate_list/jsx/candidateListIndex.js

```diff
@@ -64,7 +64,17 @@ class CandidateListIndex extends Component {
   fetchData() {
     return fetch(this.props.dataURL, {credentials: 'same-origin'})
       .then((resp) => resp.json())
-      .then((data) => this.setState({data}))
+      .then((data) => {
+        // Convert concatenated string of subproject and visit labels to array
```

Comment:
could you do all this in php/candidate_list.class.inc? manipulate result of \Utility::getVisitList() so you get an associative array i.e. 'V1'=>'V1', 'V2'=>'V2'

i personally don't think this should be handled on the client side

Reply:
I followed your implementation from `user_accounts` (#4167). 
I can make the requested change though. What do you think?

Reply:
ohh i see. i thought the array would have directly sent over but because we json'd it, i think it's still necessary here to convert it back to an array. i think this makes sense